include/cereal/details/static_object.hpp

```diff
@@ -44,7 +44,7 @@
     License, Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
     http://www.boost.org/LICENSE_1_0.txt) */
 
-#ifdef _MSC_VER
+#if defined(_MSC_VER) && !defined(__clang__)
```

Comment:
clang-cl supports declspecs, so I'd imagine this is useful as well.

Reply:
declspec causes this error on clang9

cereal::detail::StaticObject<cereal::detail::bind_to_archives<PolyDerivedLA, cereal::detail::(anonymous namespace)::polymorphic_binding_tag> >' must have external linkage when declared 'dllexport'

Reply:
this error can be reproduced on test_polymorphic

Reply:
We (myself + @fnabulsi + @bmcdb) got the identical error and had to make the same change, although I acknowledge this may warrant more investigation.

We are testing cereal only with static linking so it wouldn't surprise me if there's more work to ensure that clang-cl can properly build Cereal as a DLL.